service/history/shard/context_impl.go

```diff
@@ -1949,6 +1985,10 @@ func (s *ContextImpl) acquireShard() {
 		// we know engineFuture must be ready here, and we can notify queue processor
 		// to trigger a load as queue max level can be updated to a newer value
 		s.notifyQueueProcessor()
+		// This runs until the lifecycleCtx is cancelled, so we only need to start it once
+		if s.emittingQueueMetrics.CompareAndSwap(false, true) {
+			go s.monitorQueueMetrics()
+		}
```

Comment:
nit: sync.Once maybe ?

Reply:
Oh right, that exists. I'll make that change